src/main/scala/org/wikimedia/analytics/hdfstools/HdfsRsyncExec.scala

```diff
@@ -0,0 +1,411 @@
+// Copyright 2019 Wikimedia Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.wikimedia.analytics.hdfstools
+
+import org.apache.hadoop.fs.permission.{ChmodParser, FsPermission}
+import org.apache.hadoop.fs.{FileStatus, FileUtil, Path}
+import org.apache.log4j.{ConsoleAppender, Logger, PatternLayout}
+
+
+/**
+ * Class executing rsync given the provided config
+ *
+ * As a convention, we use dst to name the parent folder of a copy,
+ * and target the exact target of a copy in dst (with same filename part)
+ *
+ * Note: The behavior of HdfsRsync differs from the original rsync one when src is a folder and
+ * the algorithm runs without recursion: original rsync skips the folders, while we copy the folder
+ * and its content at once if it is not present, or overwrite the full folder at once if modification
+ * timestamp differs (this is equivalent to treat folders as files, with cp -r). This feature can be
```

Comment:
I think this is more equivalent to to rm $dir && cp -a

Reply:
Done.